src/components/vendorcomponents/DateInputWithLabel.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+interface DateInputProps {
+  label: string;
+  register: any;
+  name: string;
+}
+
+export const DateInputWithLabel: React.FC<DateInputProps> = ({
+  label,
+  register,
+  name,
+}) => (
+  <div>
+    <label className="block text-sm font-medium text-gray-700">{label}</label>
+    <input
+      type="date"
+      {...register(name)}
+      className="mt-1 block w-full border border-gray-300 rounded-md shadow-sm p-1"
+    />
+  </div>
```

Comment:
This is not how we make something reusable. add more props to this, such as classnames, also make the type dynamic, stop hardcoding it